### ISSUE TITLE
resize wont update tooltip and ticks

### ISSUE THREAD

**Issue by scharc** [seiyria/bootstrap-slider#341]
when the slider is inside of a container-fluid container resizing the window wont update the tooltip and tick positions. 
i tried the refresh() function, but with the ugly to-range-slider-bug, its not working

**Comment by seiyria** [comment]
Please provide a jsfiddle as per the contribution guidelines.

**Comment by Nicien** [comment]
Workaround:

    // initialize slider:
    var slider = $("my_slider_id").slider({options}).data('slider');

    // *** fix resize bug ***/
    var self = slider; // 'self' can be your controller

    // detect size change of sliderElem
    var slider_element = $(this.slider.sliderElem);
    self.last_slider_size = slider_element.width(); 
    $(window).resize(function() {
        var new_width = slider_element.width();
        if (self.last_slider_size != new_width) {
            self.last_slider_size = new_width;

            // Ugly hack here: (note slider.relayout() didn't fix the bug)
            slider.mousedown({});
            slider.mouseup({});
        }
    });

Your welcome.

**Comment by julianxhokaxhiu** [comment]
Here it is: http://jsbin.com/xeyofajiki/1/edit?html,css,js,output

You can see that if you resize the windows the slider is "recalculated" but the tooltips are not. This is a major bug imho and should be fixed ASAP.

**Comment by cesaric** [comment]
The issue with the tick labels seems to be in the _layout() function.  Particularly this  (~ line 964):

`var labelSize = this.size / (this.options.ticks.length - 1);`

**this.size** needs to be recalculated.  So if you add:

`this.size = this.sliderElem[this.sizePos];`

as the first line of the _layout() function, it should make `relayout()` work the way you expect.  

For anyone curious, this is basically how I'm making a horizontal slider mildly responsive in Bootstrap (using bootstrap-toolkit.js):

```javascript
$(window).resize(function () {
     if (viewport.is("sm")) {
             $('#basicSlider').addClass('smallSlider');
     } else {
             $('#basicSlider').removeClass('smallSlider');
     }
     basicSlider.relayout();
});
```
And don't forget the whole **data-slider-id** shenanigans if you do CSS with id selectors/media queries instead of applying them programmatically.

**Comment by julianxhokaxhiu** [comment]
@rovolution now that issue is closed means that is fixed or what?

**Comment by rovolution** [comment]
@julianxhokaxhiu could you create a fiddle with the latest version of the library and confirm whether or not this bug still occurs? thanks!

**Comment by julianxhokaxhiu** [comment]
@rovolution This link is still valid and is loading the latest version of your library. And the bug is still there :)

http://jsbin.com/xeyofajiki/1/edit?html,css,js,output

**Comment by Bhoft** [comment]
you can see a fixed version which uses code from my fork where i used positioning with transform: translate.
http://jsbin.com/dufufevuze/edit?html,css,js,output

which i also mentioned here
https://github.com/seiyria/bootstrap-slider/issues/308

**Comment by julianxhokaxhiu** [comment]
@Bhoft That's exactly what I would like to reach. Do you mind to do a PR?

**Comment by Bhoft** [comment]
I have created one but it was changing some other stuff as well.
https://github.com/seiyria/bootstrap-slider/pull/395

**Comment by evillemez** [comment]
Any news?  Is there a particular fork available with a fix for the tick label resizing?

**Comment by rbojha** [comment]
HI @Bhoft, this looks ok to some extent... when used horizontally it looks perfect, but  in vertical scenario the tick labels are not in right position... 

So, basically my question is, Any update on this??

@rovolution, any comment will be really helpful??

**Comment by Bhoft** [comment]
i have a fiddle with both vertical and horizontal sliders also with an modal
(this is using my fork)
http://jsfiddle.net/h3WDq/794/

(this the original code) 
http://jsfiddle.net/h3WDq/752/

I can't see a problem on the vertical slider. maybe you can also add a fiddle.

**Comment by KBog** [comment]
This is how I fixed it:

- My custom ID for the parent div that contains the bar and the sliders with class `slider slider-horizontal` is `search-slider-slidercontrol`.
- It's the id that goes as parameter when creating the slider:
`var sliderControl = new Slider("#search-slider-inputid", {`
`  id: "search-slider-slidercontrol",  <<<<<< this one`
  .
  .

- Newly added CSS:
```
/* This is to override the slider-tick-label-container which has a margin-left, and instead kill the margin */
#search-slider-slidercontrol .slider-tick-label-container {
  margin-left: 0 !important;
}

/* Created a new CSS for each label (min & max) */
.search-slider-slidercontrol-minlabel {
  width: auto !important;
}
.search-slider-slidercontrol-maxlabel {
  width: auto !important;
  float: right;
}
```

- To add those newly created CSS classes in Javascript, I had to inject the classes using jquery, I couldn't just fetch the min/max labels by class or id, because they use the same class:
```
// 1) a) Fetch the parent div at ID #search-slider-slidercontrol
// 1) b) Fetch the child divs with the class .slider-tick-label inside the parent div.
var labelsInDOM = $('.slider-tick-label', $('#search-slider-slidercontrol'));
// 2) Add the custom class for the first/min one (left label)
$(labelsInDOM[0]).addClass('search-slider-slidercontrol-minlabel');
// 3) Add the custom class for the second/max one (right label)
$(labelsInDOM[1]).addClass('search-slider-slidercontrol-maxlabel');
```

That's it, now you can resize the window and everything moves as expected.
Note that you can use this as is, just make sure to set the proper parent div ID to match yours.

Hope that helps,
KB

**Comment by rovolution** [comment]
Fixed in [v5.3.4](https://github.com/seiyria/bootstrap-slider/blob/master/CHANGELOG.md#534--2015-12-27)
